### PR TITLE
feat(FEC-8277): Add picture-in-picture support

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

added a picture in picture support for chrome (from version 70) + safari.

**still pending design. the icon place + design is still pending Hillel design**

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/picture-in-picture/picture-in-picture.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import {KeyMap} from '../../utils/key-map';
+import BaseComponent from '../base';
+
+/**
+ * PictureInPicture component
+ *
+ * @class PictureInPicture
+ * @extends {BaseComponent}
+ */
+class PictureInPicture extends BaseComponent {
```
Comment (dan-ziv) [inline src/components/picture-in-picture/picture-in-picture.js]: Don't create the component at all if isPictureInPictureSupported() returns false.

**Comment by odedhutzler** [comment]
@odedhutzler need to listen to enter exit chrome events and change state accordingly

File: src/components/picture-in-picture/picture-in-picture.js (hunk 1 above)
Comment (odedhutzler) [reply src/components/picture-in-picture/picture-in-picture.js]: @OrenMe

Comment (dan-ziv) [reply src/components/picture-in-picture/picture-in-picture.js]: ?

File: src/ui-presets/playback.js [hunk 2]
```diff
@@ -59,6 +60,7 @@ export function playbackUI(props: any): React$Element<any> {
             <LanguageControl player={props.player} />
             <SettingsControl player={props.player} />
             <CastControl player={props.player} />
+            <PictureInPicture player={props.player} />
```
Comment (dan-ziv) [inline src/ui-presets/playback.js]: dont create the component if PIP isn't supported

File: src/components/picture-in-picture/picture-in-picture.js (hunk 1 above)
Comment (OrenMe) [reply src/components/picture-in-picture/picture-in-picture.js]: @dan-ziv we will use this component rendering like this. 
virtual DOM knows to render or not by receiving undefined.

File: src/components/picture-in-picture/_picture-in-picture.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.picture-in-picture{
```
Comment (dan-ziv) [inline src/components/picture-in-picture/_picture-in-picture.scss]: place this inside `.player` class

Comment (odedhutzler) [reply src/components/picture-in-picture/_picture-in-picture.scss]: fixed

**Pull request merged**